src/canmatrix/canmatrix.py

```diff
@@ -435,13 +434,16 @@ def byName(self, name):
                 return test
         return None
 
-    def __str__(self):
-        return self.name
-
     def __iter__(self):
         """Iterate over all contained signals."""
         return iter(self.signals)
 
+    def __getitem__(self, name):
+        signal = self.byName(name)
+        if signal:
+            return signal
+        raise IndexError("Signal {} doesn't exist".format(name))
```

Comment:
`KeyError`?  Though it's not really either so maybe a custom exception.

Maybe `repr(name)` or otherwise get it in quotes?

Reply:
Yeah, I believe I know experienced people that prefer both ways.  I like the separation so callers can make more specific except clauses.  I might change it later, but I'll let `KeyError` go for now.  :]